src/node/txreconciliation.cpp

```diff
@@ -142,9 +250,104 @@ class TxReconciliationTracker::Impl
         return (recon_state != m_states.end() &&
                 std::holds_alternative<TxReconciliationState>(recon_state->second));
     }
+
+    // Not const because of caching.
+    bool IsFanoutTarget(const Wtxid& wtxid, NodeId peer_id, bool we_initiate, double limit) EXCLUSIVE_LOCKS_REQUIRED(m_txreconciliation_mutex)
+    {
+        auto fanout_candidates = m_tx_fanout_targets_cache_data.find(wtxid);
+        if (fanout_candidates != m_tx_fanout_targets_cache_data.end()) {
+            return std::binary_search(fanout_candidates->second.begin(), fanout_candidates->second.end(), peer_id);
+        }
+
+        // We use the pre-determined randomness to give a consistent result per transaction,
+        // thus making sure that no transaction gets "unlucky" if every per-peer roll fails.
+        CSipHasher deterministic_randomizer{m_deterministic_randomizer};
+        deterministic_randomizer.Write(wtxid.ToUint256());
```

Comment:
Looking on `CSipHasher`, given it’s a pseudo-random hash function, verified it’s well-initialized from two hidden random 64-bit seeds in `src/init.cpp` (`L1239`). Then we add a `CSipHasher` instance provided by `CConman` at `TxReconciliationTracker` initialization in `src/net_processing.cpp`. This respect the SipHash’s PRF’s requirement to initialize it with a random 128-bit key. I still wonder if in the future `TxReconciliationTracker` shouldn’t get it’s own random seed (i.e use `GetRand()`, it promises fast entropy generation) to isolate tx-announcement from the rest of network connection management.

Reply:
This seems to be consistent with how a deterministic randomizer is seeded in many other places in the codebase. What is your rationale for making it different here?